Lib/test/test_math.py

```diff
@@ -1948,75 +1946,74 @@ def testComb(self):
-    #     self.assertEqual(math.ulp(2 ** 53), 2.0)
-    #     self.assertEqual(math.ulp(2 ** 64), 4096.0)
-
-    #     # min and max
-    #     self.assertEqual(math.ulp(0.0),
-    #                      sys.float_info.min * sys.float_info.epsilon)
-    #     self.assertEqual(math.ulp(FLOAT_MAX),
-    #                      FLOAT_MAX - math.nextafter(FLOAT_MAX, -INF))
-
-    #     # special cases
-    #     self.assertEqual(math.ulp(INF), INF)
-    #     self.assertIsNaN(math.ulp(math.nan))
-
-    #     # negative number: ulp(-x) == ulp(x)
-    #     for x in (0.0, 1.0, 2 ** 52, 2 ** 64, INF):
-    #         with self.subTest(x=x):
-    #             self.assertEqual(math.ulp(-x), math.ulp(x))
+    @requires_IEEE_754
+    def test_nextafter(self):
+        # around 2^52 and 2^63
+        self.assertEqual(math.nextafter(4503599627370496.0, -INF),
+                         4503599627370495.5)
+        self.assertEqual(math.nextafter(4503599627370496.0, INF),
+                         4503599627370497.0)
+        self.assertEqual(math.nextafter(9223372036854775808.0, 0.0),
+                         9223372036854774784.0)
+        self.assertEqual(math.nextafter(-9223372036854775808.0, 0.0),
+                         -9223372036854774784.0)
+
+        # around 1.0
+        self.assertEqual(math.nextafter(1.0, -INF),
+                         float.fromhex('0x1.fffffffffffffp-1'))
+        self.assertEqual(math.nextafter(1.0, INF),
+                         float.fromhex('0x1.0000000000001p+0'))
+
+        # x == y: y is returned
+        self.assertEqual(math.nextafter(2.0, 2.0), 2.0)
+        self.assertEqualSign(math.nextafter(-0.0, +0.0), +0.0)
+        self.assertEqualSign(math.nextafter(+0.0, -0.0), -0.0)
+
+        # around 0.0
+        smallest_subnormal = sys.float_info.min * sys.float_info.epsilon
+        self.assertEqual(math.nextafter(+0.0, INF), smallest_subnormal)
+        self.assertEqual(math.nextafter(-0.0, INF), smallest_subnormal)
+        self.assertEqual(math.nextafter(+0.0, -INF), -smallest_subnormal)
+        self.assertEqual(math.nextafter(-0.0, -INF), -smallest_subnormal)
+        self.assertEqualSign(math.nextafter(smallest_subnormal, +0.0), +0.0)
+        self.assertEqualSign(math.nextafter(-smallest_subnormal, +0.0), -0.0)
+        self.assertEqualSign(math.nextafter(smallest_subnormal, -0.0), +0.0)
+        self.assertEqualSign(math.nextafter(-smallest_subnormal, -0.0), -0.0)
+
+        # around infinity
+        largest_normal = sys.float_info.max
+        self.assertEqual(math.nextafter(INF, 0.0), largest_normal)
+        self.assertEqual(math.nextafter(-INF, 0.0), -largest_normal)
+        self.assertEqual(math.nextafter(largest_normal, INF), INF)
+        self.assertEqual(math.nextafter(-largest_normal, -INF), -INF)
+
+        # NaN
+        self.assertIsNaN(math.nextafter(NAN, 1.0))
+        self.assertIsNaN(math.nextafter(1.0, NAN))
+        self.assertIsNaN(math.nextafter(NAN, NAN))
+
+    @requires_IEEE_754
```

Comment:
This won't actually be run because of the stub impl of `requires_IEEE_754` -- you could try changing it in `support/__init__.py` to `requires_IEEE_754 = unittest.skipIf(False, "RustPython always has IEEE 754 floating point numbers")`

Reply:
Silly me. I commented out the lines in the tests that failed. I both perform operations on the constant `sys.float_info.min * sys.float_info.epsilon`. I wonder if RustPython sets these constants incorrectly or something?

Reply:
Oh, yeah, I think you're right. I think the `min: f64::MIN` in `sysmodule.rs` should actually be `min: f64::MIN_POSITIVE`

Reply:
Thanks. That got the test passing.